ingest/validate_de.py

```diff
@@ -0,0 +1,141 @@
+import pandas as pd
+import itertools
+import numpy as np
+file_path = "/Users/mvelyuns/scp-ingest-pipeline/ingest/pairwise.csv"
+data = pd.read_csv(file_path)
+
+genes = data.iloc[:, 0].tolist()
+rest = data[data.columns[1:]]
+col = list(rest.columns)
+
+col_values = {}
+for i in col:
+    col_values[i] = rest[i].tolist()
+
+#note: my initial files had pval, qval, log2fc. David's files have qval, mean, log2fc. For the purposes of this validation I will be using his column values/formatting. If we choose to remove mean again this code can be changed
+
+#question: should user also input type names to not rely solely on parsing?
+
+#below parsing relies on this format: 'type_0'_'type_1'_qval
+#necessary to include quotes
+clean_val = []
+
+#cleans column header of unnecessary characters and isolates column name, returns clean_val containing only cleaned values
+for i in col:
+    str = i.split("'")
+    col_names = []
+    for j in str:
+        if (j != '') and (j != "_"):
+            col_names.append(j.strip("_"))
+    clean_val.append(col_names)
+
+types = []
+qual = []
+
+#isolate the types and values in submitted file
+#expected format: 
+#types: ['type_0', 'type_1', 'type_2', 'type_3']
+#qual: ['qval', 'log2fc', 'mean']
+for i in clean_val:
+    if i[0] not in types:
+        types.append(i[0])
+    if i[1] not in types:
+        types.append(i[1])
+    if i[2] not in qual:
+        qual.append(i[2])
+
+#verify all pairwise combinations are present and have each metric (qval, log2fc, mean)
+#correct pairwise combinations:
+correct_pairs = list(itertools.combinations(types,2))
+
+correct_list = []
+for i in correct_pairs:
+    q1 = [i[0], i[1], qual[0]]
+    q2 = [i[0], i[1], qual[1]]
+    q3 = [i[0], i[1], qual[2]]
+    correct_list.append(q1)
+    correct_list.append(q2)
+    correct_list.append(q3)
+
+if correct_list != clean_val:
+    raise ValueError("incorrect inputted values")
+
+
+#create individual files
+#desired format:
+#for ex, if we have 
+#'type_0'_'type_1'_qval
+#'type_0'_'type_1'_log2fc
+#'type_0'_'type_1'_mean
+#final format should have type 0 type 1 in the title, and genes, qval, log2fc, mean as columns
+names_dict = {}
+all_group = []
+for i in range(len(types)):
+    curr_type = types[i]
+    type_group = []
+
+    for j in range(len(clean_val)):
+        curr_val = clean_val[j][0]
+        comp_val = clean_val[j][1]
+        file_naming = "'" + curr_val + "'" + "_" + "'" + comp_val + "'"
```

Comment:
This will leave `'` characters in the filenames, which is not ideal.  We can safely omit them which will give us more readable filenames.  Also, `f` strings make this much simpler:
```suggestion
        file_naming = f"{curr_val}_{comp_val}"
```